### ISSUE TITLE
[terra-action-header] Add ability to disable back and close buttons. Add 'Done' button close option.

### ISSUE THREAD

**Issue by bbaraban** [cerner/terra-core#2951]
# Feature Request

## Description
I've been working on getting our components off the terra-clinical-header and onto the terra-action-header. I've run into some issues that I would like to request enhancements for.

1. The first is the ability to indicate that the back button and close button should be disabled. We have specific work flows where we do not want the user to be able to use these buttons while certain conditions are met. Most notably is when orders are signing we don't want them to leave the view as important information may be disabled once signing is done.

There is a few current options, but I find all less than ideal, for example I can short-circuit the logic or just not pass a method to terra-action-header. Though the later will not render the button result buttons disappearing and reappearing.

2. The second request is more complicated but is still an issue for us. We have workflows, specifically in slide out pane were a user can edit order details, we use a Button with the label 'Done' as oppose to an X close button. We received feedback that the user feels that the X close button may discard their changes where a Done button provides more confidence there changes were saved.

Currently I've work around this by just not providing a onClose method and including my own Done button next to my collapsible menu item. Though not supported according terra-action-header API doc it works well. Like the X button we don't won't Done to ever collapse so that is why I can't include in the collapsible collection.

There is a few options I can think of that can help achieve this.

- Add indicator to prevent a collapsible menu items from collapsing. This would allow us to add the Done button to the collection and have it never be collapsible. 
- Enhance terra-action-header to flex onClose X button with a traditional button and custom label

## Additional Context / Screenshots
Below is a mock up of when and what a disabled back button would look like. This would present right after the user clicked Sign.
![Screen Shot 2020-05-05 at 8 15 16 AM](https://user-images.githubusercontent.com/34078628/81073060-af03c380-8eac-11ea-87cc-c846b199e599.png)

Here is an example on when we use a Done button instead of an X button.
![Screen Shot 2020-05-05 at 8 48 57 AM](https://user-images.githubusercontent.com/34078628/81073669-9b0c9180-8ead-11ea-8b3e-172c01385a71.png)
## @ Mentions

@tbiethman

**Comment by neilpfeiffer** [comment]
Closing as not supportable requests by Terra.

Answering your request in the  two parts in which they were asked:
> 1. Add ability to disable back and close buttons

Selectively disabling buttons in the action-header is not a pattern UX recommends. If the contents of your page/panel/modal has content that requires in the interactions of the entire screen be locked/blocked from the user taking further actions, a better way to implement blocking user interactions while waiting on a synchronous action is with a loading overlay to cover over the entire page/modal (headers included). If it is also a danger that user has the ability to navigate away using any of the application navigation items (or any other active UI or navigation active to a user), additional discussions may need to be explored for an application approach, though nothing like that exists yet.

> 2. Add 'Done' button close option

Per UX Design Standards, the [action-header](https://wiki.cerner.com/x/tJjGZQ) is not permitted to have a "done" option in the place of the "x" close action. Recommendation is to work with a designer to update your layouts from the prior mobile/iOS inspired layout patterns to more current universal Cerner design patterns that puts buttons for terminal actions in the [action-footer](https://wiki.cerner.com/x/upPLZg). Also worth revisiting is the button label `Done`, since it is unclear to a user what "Done" means and generally not recommended best usability practice.

**Comment by neilpfeiffer** [comment]
Created a new issue to investigate options for an application-level overlay related to the suggestion/comment I gave the first request item:

> 1. Add ability to disable back and close buttons
[...] If it is also a danger that user has the ability to navigate away using any of the application navigation items (or any other active UI or navigation active to a user), additional discussions may need to be explored for an application approach, though nothing like that exists yet.

Terra can investigate to see if these types of system/service requests might best be handled by a full-screen all application loading overlay: https://github.com/cerner/terra-framework/issues/1112
